lib/logger/mix.exs

```diff
@@ -20,7 +20,7 @@ defmodule Logger.MixProject do
         backends: [:console],
         translators: [{Logger.Translator, :translate}],
         sync_threshold: 20,
-        discard_threshold: 500,
+        discard_threshold: 10_000,
```

Comment:
This change could lead to 9980 processes blocking. If we are to provide this stronger guarantee on message queue length should we also set sync threshold to `10_000` by default? I don't think we really need `sync` any more unless desired? Also we don't want to block for the time it takes to handle `9999` events. This would be very bad buffer bloat for events not discarded.

Reply:
I  think we need to bump the current threshold but maybe `10_000` is way too hight. Would 2000 alleviate the issue?

Reply:
Reverted.